src/kani-compiler/rustc_codegen_kani/src/codegen/statement.rs

```diff
@@ -436,17 +436,14 @@ impl<'tcx> GotocCtx<'tcx> {
         let assert_nonnull = Stmt::assert(call_is_nonnull, &assert_msg, loc.clone());
 
         // Virtual function call and corresponding nonnull assertion.
-        let call = if self.vtable_ctx.emit_vtable_restrictions {
-            self.virtual_call_with_restricted_fn_ptr(
-                trait_fat_ptr.typ().clone(),
-                idx,
-                fn_ptr,
-                fargs.to_vec(),
-            )
+        let call = fn_ptr.dereference().call(fargs.to_vec());
+        let call_stmt = self.codegen_expr_to_place(place, call).with_location(loc.clone());
```

Comment:
The changes do not seem to preserve the previous behavior: previously it was calling `self.codegen_expr_to_place` on `call` (which is the restricted function pointer if vtable restrictions is enabled), but now it is calling `self.virtual_call_with_restricted_fn_ptr` on the return value of `self.codegen_expr_to_place`. Is this intentional?

Reply:
If you'd like a more definitive answer, I could ping Alexa about it. But I believe the answer is yes.

It was previously translating the function call (`foo(...)`) to a new function (calling a generated wrapper function) call expression. Now it's doing the assignment (`place = foo(...)`) but inserting a label beforehand to reference the call site. The label has to be at the statement-level, rather than before where we were transforming something at the expression level.

Notice that `virtual_call_with_restricted_fn_ptr` changed from `Expr` to `Stmt` return type.